core/common/src/main/java/alluxio/wire/TieredIdentity.java

```diff
@@ -58,9 +64,37 @@ public TieredIdentity(List<LocalityTier> tiers) {
    * @return the corresponding wire type tiered identity
    */
   public static TieredIdentity fromThrift(alluxio.thrift.TieredIdentity tieredIdentity) {
+    if (tieredIdentity == null) {
+      return null;
+    }
     return new TieredIdentity(tieredIdentity.getTiers().stream()
-        .map(LocalityTier::fromThrift).collect(Collectors.toList())
-    );
+        .map(LocalityTier::fromThrift).collect(Collectors.toList()));
+  }
+
+  /**
+   * @param identities the tiered identities to compare to
+   * @return the identity closest to this one
```

Comment:
should comment on when none of them match, or what `empty` represents.

Reply:
done